### PR TITLE
Navigation drawer overhaul

### PR DESCRIPTION
**Pull request by karllindmark**
Currently includes:
- Refactoring of Navigation Drawer implementation
- All previously searched soldiers are now accessible in the dropdown
- LoginActivity now only returns ONE soldier (matched on personaId+platformId)

TODO:
- Decide what to show on HomeFragment
- Make the Navigation Drawer display the current menu selection state properly (maybe change background resource?)
- Update to use v7 version of ActionbarDrawerToggle (+Toolbar so that the navigation drawer slides over it)
- Animations for the menu toggling (when you show/hide the soldiers) ala I/O?

[WIP] @peter-budo

### CODE REVIEW

**Comment by karllindmark** [comment]
TODO:
- [x] Display relevant information on `HomeFragment`
- [x] Display the current selection state in the `NavigationDrawerFragment`
- [x] Animate the soldier toggling in the `NavigationDrawerFragment` (when clicking on the soldier box)?

File: BF4Intel/src/main/java/com/ninetwozero/bf4intel/database/migrations/MigrationToOneZeroSix.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import com.ninetwozero.bf4intel.database.dao.login.SummarizedSoldierStatsDAO;
+
+public class MigrationToOneZeroSix extends BaseMigration {
+
+    @Override
+    protected void doMigration(SQLiteDatabase db) {
+        dropTable(db, SummarizedSoldierStatsDAO.TABLE_NAME);
+        createNewSummarizedSoldierStatsTable(db);
+        // (SummarizedSoldierStatsDAO);
```
Comment (peter-budo) [inline BF4Intel/src/main/java/com/ninetwozero/bf4intel/database/migrations/MigrationToOneZeroSix.java]: Comment, please remove

File: BF4Intel/src/main/res/layout/list_item_navdrawer_separator.xml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<View xmlns:android="http://schemas.android.com/apk/res/android"
+      android:layout_marginTop="7dp"
```
Comment (peter-budo) [inline BF4Intel/src/main/res/layout/list_item_navdrawer_separator.xml]: Hard coded

File: BF4Intel/src/main/res/layout/list_item_navdrawer_soldier.xml [hunk 3]
```diff
@@ -32,6 +24,20 @@
     android:textAppearance="@android:style/TextAppearance.Holo.Small.Inverse"
     android:textSize="@dimen/text_s"
     android:textAllCaps="true"
+    tools:text="PS4"
+    />
+
+  <ImageView
+    android:id="@+id/soldier_dropdown_arrow"
+    android:layout_width="32dp"
```
Comment (peter-budo) [inline BF4Intel/src/main/res/layout/list_item_navdrawer_soldier.xml]: Hard coded

File: BF4Intel/src/main/res/layout/list_item_navdrawer_soldier_dropdown.xml [hunk 4]
```diff
@@ -32,6 +24,19 @@
     android:textAppearance="@android:style/TextAppearance.Holo.Small.Inverse"
     android:textSize="@dimen/text_s"
     android:textAllCaps="true"
+    tools:text="PS4"
+    />
+
+  <ImageView
+    android:id="@+id/soldier_dropdown_arrow"
+    android:layout_width="32dp"
```
Comment (peter-budo) [inline BF4Intel/src/main/res/layout/list_item_navdrawer_soldier_dropdown.xml]: Hardcoded

File: BF4Intel/src/main/res/values/content_strings.xml [hunk 5]
```diff
@@ -36,4 +36,5 @@
   <string name="cdesc_reward_image">Reward image</string>
   <string name="cdesc_completion_indicator">Completion indicator</string>
   <string name="cdesc_expansion_icon">Expansion icon</string>
+  <string name="cdesc_arrow">Arrow</string>
```
Comment (peter-budo) [inline BF4Intel/src/main/res/values/content_strings.xml:39]: Arrow has no meaning to blind person "Back" has more meaning

Comment (karllindmark) [inline BF4Intel/src/main/res/values/content_strings.xml:39]: It's for the toggling purposes, so maybe "Toggle"?

**Comment by peter-budo** [comment]
Two list items selected

Basically after fresh install, you search for soldier, soldier selected and you are redirected to Overview. Then you select any option from navigation draw. On next open of navigational drawer two list items are marked as selected.

This is because when search for soldier is done, user is redirected to overview, however `formatNavigationDrawerItem(View view, boolean selected)` is not called on `Home` to deselect and on `Overview` to select :wink: 

device-2014-11-03-201923

**Comment by peter-budo** [comment]
The only two outstanding issues I can see is holo theme on dialog and navigation drawer icon position changing on selection/desection

**Comment by karllindmark** [comment]
Last issue is only on 7''+?  Or sw720dp?

**Comment by peter-budo** [comment]
@karllindmark ping me when you around

Problems that I noticed so far:
- [x] HDPI phone not showing navigation drawer image
- [x] HDPI phone & 10" tablet no navigation drawer left padding
- [x] Action bar back arrow navigation from details to parent activity messed up it goes to Overview, example, open Assignments, select an assignment to see. Once there press arrow on top and you end up in Overview. NOTE: Back button is fine, it is just AB arrow that is misleading

**Pull request merged**